src/main/kotlin/com/hedvig/underwriter/service/UnderwriterImpl.kt

```diff
@@ -143,13 +143,15 @@ class UnderwriterImpl(
     ): MutableList<BreachedGuidelineCode> {
         val guidelineErrors = mutableListOf<BreachedGuidelineCode>()
 
-        listOfRules.forEach {
-            val error = it.invokeValidate(data)
-            if (error != null) {
-                guidelineErrors.add(error)
-            }
-            if (it.skipAfter) {
-                return@forEach
+        run validateGuidelines@{
+            listOfRules.forEach {
+                val error = it.invokeValidate(data)
+                if (error != null) {
+                    guidelineErrors.add(error)
+                    if (it.skipAfter) {
+                        return@validateGuidelines
+                    }
+                }
```

Comment:
Wouldn't it be simpler to read if this were just a:
```suggestion
        for (rule in listOfRules) {
            val error = it.invokeValidate(data)
            if (error != null) {
                guidelineErrors.add(error)
                if (it.skipAfter) {
                    break
                }
            }
```
I feel the extra `run` and labeling feels like a double-dance to avoid a simple for loop?

Reply:
Agreed the @ concept is maybe not the best :slightly_smiling_face: